docker/sds-api/spine-directory-service/sds/request/cpm.py

```diff
@@ -63,76 +128,90 @@ def handle_error(response, call_name):
         raise SDSException(detail)
 
 class CpmClient:
-    def __init__(self, client_id: str, apigee_url: str,  endpoint: str) -> None:
+    FILTER_MAP = {}
+    def __init__(self, client_id: str, apigee_url: str, query_params: dict) -> None:
         self._client_id = client_id
         self._apigee_url = apigee_url
-        self._endpoint = endpoint
-
+        self._params = self._set_params(query_params)
+        
     async def get_cpm(self, extra_headers: dict):
         logger.info("Contacting CPM")
         url = f"https://{self._apigee_url}"
-        search_endpoint = f"Device?device_type={self._endpoint}&use_mock=true"
+        search_endpoint = self._endpoint
         headers = {
             'version': '1',
             'Authorization': 'letmein',
             'Content-Type': 'application/json',
-            'apiKey': self._client_id,
+            'apikey': self._client_id,
             **extra_headers
         }
-        params = {}
-        logger.info("Requesting data from... {url}/{endpoint}", fparams={"url": url, "endpoint": search_endpoint})
-        res = make_get_request(call_name="SDS get_cpm", url=f"{url}/{search_endpoint}", headers=headers, params=params)
+        logger.info("Requesting data from... {url}/{endpoint}", fparams={"url": url, "endpoint": search_endpoint,  "query_params": self._params})
+        res = make_get_request(call_name="SDS get_cpm", url=f"{url}/{search_endpoint}", headers=headers, params=self._params)
         return self._get_response(res=res)
 
+    def _set_params(self, query_params: Dict[str, str]) -> Dict[str, str]:
+        params = {self.FILTER_MAP.get(key, key): value for key, value in query_params.items() if value is not None}
+        return params
+
     def _get_response(self, res):
         return res.json()
 
-class BaseCpm:
-    FILTER_MAP = {}
-    DATA_MAP = {}
-    DEFAULT_DICT = {}
+class DeviceClient(CpmClient):
```

Comment:
Minor point but calling this CPMDeviceClient might make it less confusing in other spots - same with the EndpointClient